plugin/trino-hive/src/main/java/io/trino/plugin/hive/InternalHiveConnectorFactory.java

```diff
@@ -78,10 +78,10 @@ private InternalHiveConnectorFactory() {}
 
     public static Connector createConnector(String catalogName, Map<String, String> config, ConnectorContext context, Module module)
     {
-        return createConnector(catalogName, config, context, module, Optional.empty());
+        return createConnector(catalogName, config, context, module, Optional.empty(), Optional.empty());
     }
 
-    public static Connector createConnector(String catalogName, Map<String, String> config, ConnectorContext context, Module module, Optional<HiveMetastore> metastore)
+    public static Connector createConnector(String catalogName, Map<String, String> config, ConnectorContext context, Module module, Optional<HiveMetastore> metastore, Optional<Module> cachingDirectoryListerModule)
```

Comment:
`Optional<CachingDirectoryLister> cachingDirectoryLister` (like we do for metastore)
having the test provide a module (vs instance) is not needed

Reply:
Good catch. It is indeed much better to encapsulate the configuration logic in the `CachingDirectoryListerModule` and avoid letting the tests dealing with such low level details.